tools/moth.py

```diff
@@ -328,15 +334,35 @@ def __init__(self, path, seed):
         self.path = path
         self.seed = seed
         self.pointvals = []
+        self.catmod = None
+
+        try:
+            catmod = SourceFileLoader(
+                'catmod',
+                os.path.join(path, 'category.py')).load_module()
+            assert all([
+                hasattr(catmod, 'make'),
+                hasattr(catmod, 'points'),
+                type(catmod.points) is list,
+            ])
+            self.catmod = catmod
+            self.pointvals.extend(catmod.points)
+        except:
+            pass
+
         for fpath in glob.glob(os.path.join(path, "[0-9]*")):
             pn = os.path.basename(fpath)
             points = int(pn)
             self.pointvals.append(points)
+
         self.pointvals.sort()
 
     def puzzle(self, points):
-        path = os.path.join(self.path, str(points))
-        return Puzzle(self.seed, path)
+        if self.catmod is not None and points in self.catmod.points:
+            return Puzzle(self.seed, points=points, category=self.catmod)
```

Comment:
This feels like a cumbersome design; I'd like puzzles to not need to know what category they're in. Consider:

```python
if self.catmod:
    p = Puzzle(self.seed, points=points)
    catmod.make(p, points)
```